src/grid.ts

```diff
@@ -0,0 +1,146 @@
+﻿/// <reference path="../scripts/typings/jquery/jquery.d.ts" />
+
+module iq {
+	export module onboard {
+
+		/*
+		* Generic tabulated data handling class.
+		* Provides functionality for connecting the specific API for the onboarding project
+		* and loading the records into a given html table
+		*/
+		export class Grid {
+
+			constructor(public table: string) {
+				this.rowFactory = new RowFactory();
+				this.redraw();
+			};
+
+			/*
+			* Remeasure the table constraints and refresh data display
+			*/
+			redraw(): void {
+				this.loadColumnHeaders();
+			}
+
+			/*
+			* Switches to the page which is [offset] pages from the current page
+			*/
+			pageBy(offset: number): number {
+				this.pageTo(this.page + offset);
+				return this.page;
+			};
+
+			/*
+			* Switch to the first page
+			*/
+			pageToFirst(): void {
+				this.pageTo(0);
+			};
+
+			/*
+			* Switch to the last page
+			*/
+			pageToLast(): void {
+				this.pageTo(this.numberOfPages-1);
+			};
+
+			/*
+			* Sets the zero based page number and updates the grid data
+			*/
+			pageTo(page: number): void {
+				if (page >= this.numberOfPages) {
+					console.log("Cant page to " + (page+1) + " because there are only " + this.numberOfPages + " pages!");
+					page = this.numberOfPages-1;
+				}
+				if (page < 0) {
+					console.log("Page number must be positive!");
+					page = 0;
+				}
+
+				console.log("Page: " + page);
```

Comment:
This is fine for debugging, but for some versions of IE the console is disabled if the browser is not in developer mode and calling `console.log` would cause an exception. We have a wrapper in the IMQS code, `iq.log`, that checks if `window.console` is defined before using the function. You need to do something similar if you want to be safe.